### PR TITLE
Capitalize some function and variable names to make them public

### PR DESCRIPTION
**Pull request by zli82016**
Capitalize some function and variable names to make them public. These functions are called in files  [bootstrap_utils_test.go](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/bootstrap_utils_test.go) and [provider_test.go.erb](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/provider_test.go.erb). In the next PR, these files will be moved to acctest package.

Functions and variables are listed as follows:
* sendRequestWithTimeout
* sendRequest
* config.
  *   userAgent
  *  client
  *  requestBatcherServiceUsage
 * batchingConfig
   *   sendAfter
   *   enableBatching
* xxxxOperationWaitTime
* xxxxOperationWaitTimeWithResponse
* isGoogleApiErrorWithCode
* iamPolicyVersion
* retryTimeDuration
* prefixedProject
* stringToFixed64
* listCurrentlyEnabledServices
* enableServiceUsageProjectServices
* isSqlOperationInProgressError
* skipIfEnvNotSet
* mergeBindings
* kmsCryptoKeyId
* defaultBatchSendIntervalSec
* batchingConfig
*  removeBasePathVersion

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by zli82016** [comment]
In this PR, I make the field UserAgent and Client of Config public. 

However, there are functions UserAgent and Client to get the field when the field was private.  https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/validator/getconfig.go

It is not working to just remove the two functions, as the functions are used in https://github.com/GoogleCloudPlatform/terraform-validator/blob/main/ancestrymanager/ancestrymanager.go


I am going to first rename the two functions in a separate PR to not break the validator.

File: mmv1/third_party/terraform/tests/resource_google_project_test.go [hunk 1]
```diff
@@ -13,13 +13,6 @@ import (
 	"github.com/davecgh/go-spew/spew"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
-	"google.golang.org/api/cloudresourcemanager/v1"
-)
-
-var (
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_google_project_test.go:19]: Can we make this change in a separate PR? It's hard to follow where these values went in this one.

File: mmv1/third_party/terraform/resources/resource_tags_location_tag_bindings.go.erb [hunk 2]
```diff
@@ -101,7 +101,7 @@ func resourceTagsLocationTagBindingCreate(d *schema.ResourceData, meta interface
 	// identity fields and d.Id() before read
 	
 	var opRes map[string]interface{}
-	err = tagsLocationOperationWaitTimeWithResponse(
+	err = TagsOperationWaitTimeWithResponse(
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_tags_location_tag_bindings.go.erb:104]: This looks like it's calling a different method now?

**Review by rileykarson — COMMENTED** [review]
Found out how to do a case insensitive diff per https://stackoverflow.com/questions/17380029/how-to-perform-case-insensitive-diff-in-git:

```
git config --global difftool.idiff.cmd 'diff -i $LOCAL $REMOTE'
git difftool -y --tool idiff c6370e70fe5dd96a6b27f9e9421460bd36de78ff upstream/acctest-pre-1
```

Reviewers will need to change the base commit in the event of a rebase/merge on this branch- it should be the latest merge point of upstream/main.

Couple early notes inline- not a complete review pass yet.

File: mmv1/third_party/terraform/tests/resource_google_project_test.go (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/tests/resource_google_project_test.go:19]: Sure. That makes sense.

Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_google_project_test.go:19]: For pname in a separate PR we could also just eliminate it altogether and use the project id across-the-board.

File: mmv1/third_party/terraform/resources/resource_tags_location_tag_bindings.go.erb (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/resources/resource_tags_location_tag_bindings.go.erb:104]: Good catch. Thanks.

**Comment by rileykarson** [comment]
Also note the failing build:

```
google-beta/bootstrap_utils_test.go:530:51: config.userAgent undefined (type *Config has no field or method userAgent, but does have UserAgent)
google-beta/bootstrap_utils_test.go:561:20: undefined: mergeBindings
```

**Review by rileykarson — APPROVED** [review]
LGTM- human-reviewed portions of this change are contained in https://gist.github.com/rileykarson/6a44ccb070947bfef722da3aaa7e56fa (as well as limited spot checking). Otherwise I'm relying on the green tests to verify this works.

If we could keep subsequent changes smaller that'd be ideal. +- 500-1000 lines is a decent upper limit, where this is +- 3000.

**Comment by rileykarson** [comment]
Note: The changes from a276408a34419448cf5b76851cbad4039f9f1946 to 9fefc882b5c512ee0e480e3fd9fc62c9d96b230f were immaterial, I requested we merge after passing unit tests.
